app/repositories/SessionRepository.scala

```diff
@@ -89,4 +90,16 @@ class SessionRepository @Inject() (
       .deleteOne(byId(id))
       .toFuture()
       .map(_ => true)
+
+  def clearData(id: String): Future[Boolean] =
+    collection
+      .updateOne(
+        filter = byId(id),
+        update = Updates.combine(
+          Updates.set("data", Document()),
```

Comment:
Not sure this is nuanced enough for those bits of data that depend on record id - we decided to store it all by record id because we thought multiple people might use it or different items will be in progress, but this just wipes it all whenever anything finishes?

Reply:
Agreed, this needs to be specific to the journey i.e. delete all pages that get populated in the create profile journey if we're in the create profile CYA. We probably need specific clearData functions for each journey, I suppose instead of update you can just do a delete

Reply:
you could have an enum which holds pages for each journey and match on that in the clearData, loop through the pages held in a list and delete each


Looks like we already have an object for that which could be extended:

```
sealed trait Journey

case object CreateRecordJourney extends Journey {
 val pages = Seq(CommodityCodePage, ...)
  override def toString: String = "CreateRecord"
}
case object UpdateRecordJourney extends Journey {
   deg pages(recordId) = Seq(CommodityCodePage(recordId, ...)
  override def toString: String = "UpdateRecord"
}



def clearData(journey: Journey) = journey match {
case UpdateRecordJourney => UpdateRecordJourney.pages.map(page => collection.delete(page))
```

Reply:
I like the idea of extending these 🙂

Reply:
refactored to be very specific to the journey!